File: integration/maven-demo/src/test/java/demo/AppTest.java
package demo;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class AppTest {
    @Test void addWorks() { assertEquals(5, App.add(2, 3)); }

    // WU-LPR-063 failure path: -Ddemo.broken=true requires a deliberate failure.
    @Test void deliberateBreakSwitch() {
        String broken = System.getProperty("demo.broken");
        System.out.println("demo.broken=[" + broken + "]");
        assertTrue(!"true".equals(broken), "demo.broken=true requested a failing build");
    }
}

File: integration/gradle-demo/src/test/kotlin/demo/AppTest.kt
package demo

import kotlin.test.Test
import kotlin.test.assertEquals

class AppTest {
    @Test fun `add works`() { assertEquals(5, add(2, 3)) }
    @Test fun `add is commutative`() { assertEquals(add(1, 2), add(2, 1)) }

    // WU-LPR-062 failure path: -Pdemo.broken=true injects a deliberate failure
    // so the pipeline failure route can be exercised end to end.
    @Test fun `deliberate break switch`() {
        val broken = System.getProperty("demo.broken") == "true"
        if (broken) throw AssertionError("demo.broken=true requested a failing build")
    }
}
